crates/common/utils.rs

```diff
@@ -0,0 +1,35 @@
+use ethereum_types::U256;
+
+/// Converts a big endian slice to a u256, faster than `u256::from_big_endian`.
+pub fn u256_from_big_endian(slice: &[u8]) -> U256 {
+    let mut padded = [0u8; 32];
+    padded[32 - slice.len()..32].copy_from_slice(slice);
+
+    let mut ret = [0; 4];
+
+    let mut u64_bytes = [0u8; 8];
+    for i in 0..4 {
+        u64_bytes.copy_from_slice(&padded[8 * i..(8 * i + 8)]);
+        ret[4 - i - 1] = u64::from_be_bytes(u64_bytes);
+    }
+
+    U256(ret)
+}
+
+/// Converts a constant big endian slice to a u256, faster than `u256::from_big_endian` and `u256_from_big_endian`.
+///
+/// Note: N should not exceed 32.
```

Comment:
Don't just "note". Use a compile-time assertion:
```rust
const { assert!(N <= 32) };
```

Reply:
You can also add that as the error message with the normal `assert!` macro syntax.